rotkehlchen/exchanges/bitfinex.py

```diff
@@ -654,23 +641,29 @@ def _query_currency_map(self) -> CurrencyMapResponse:
                     f'{self.name} currency map returned invalid JSON response. Check further logs',
                 )
             else:
-                currency_map = {
-                    bfx_symbol: symbol for bfx_symbol, symbol in response_list[0]
-                    if bfx_symbol not in set(BITFINEX_EXCHANGE_TEST_ASSETS)
-                }
-                with GlobalDBHandler().conn.read_ctx() as cursor:
-                    cursor.execute(
-                        'SELECT exchange_symbol, local_id FROM location_asset_mappings WHERE location IS ? OR location IS NULL;',  # noqa: E501
-                        (Location.BITFINEX.serialize_for_db(),),
-                    )
-                    for bfx_symbol, asset_id in cursor:
-                        currency_map[bfx_symbol] = asset_id
+                test_assets = set(BITFINEX_EXCHANGE_TEST_ASSETS)
+                bitfinex_db_serialized = Location.BITFINEX.serialize_for_db()
+                for bfx_symbol, symbol in response_list[0]:
+                    with GlobalDBHandler().conn.read_ctx() as cursor:
+                        if cursor.execute(
+                            'SELECT COUNT(*) FROM location_unsupported_assets WHERE location=? AND exchange_symbol=?',  # noqa: E501
+                            (bitfinex_db_serialized, bfx_symbol),
+                        ).fetchone()[0] > 0 or bfx_symbol in test_assets:
+                            continue
+
+                    try:
+                        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                            write_cursor.execute(
+                                'INSERT OR IGNORE INTO location_asset_mappings (location, exchange_symbol, local_id) VALUES (?, ?, ?);', (  # noqa: E501
+                                    bitfinex_db_serialized,
+                                    bfx_symbol,
+                                    symbol_to_asset_or_token(symbol).serialize(),
+                                ),
+                            )
+                    except UnknownAsset:
+                        log.info(f'Found new asset symbol {bfx_symbol} for {symbol} in Bitfinex. Support for it has to be added.')  # noqa: E501
 
-        return CurrencyMapResponse(
-            success=was_successful,
-            response=response,
-            currency_map=currency_map,
-        )
+        return CurrencyMapResponse(success=was_successful, response=response)
```

Comment:
Can't you raise remoterror here and remove the usage where you call this method? This allows you to get rid of `CurrencyMapResponse`. Just be sure to use the same text that we have where you use this method